packages/ruleset-bundler/src/index.ts

```diff
@@ -8,14 +8,23 @@ export type BundleOptions = {
   target: 'node' | 'browser' | 'runtime';
   format?: 'esm' | 'commonjs' | 'iife';
   treeshake?: boolean; // false by default
+  fullOutput?: boolean;
 };
 
 export { IO } from './types';
 
 export async function bundleRuleset(
   input: string,
-  { target = 'browser', plugins, format, treeshake = false }: BundleOptions,
-): Promise<string> {
+  opts: Omit<BundleOptions, 'fullOutput'> | (Omit<BundleOptions, 'fullOutput'> & { fullOutput: false }),
+): Promise<string>;
+export async function bundleRuleset(
+  input: string,
+  opts: Omit<BundleOptions, 'fullOutput'> & { fullOutput: true },
+): Promise<RenderedChunk>;
+export async function bundleRuleset(
+  input: string,
+  { target = 'browser', plugins, format, treeshake = false, fullOutput = false }: BundleOptions,
+): Promise<string | RenderedChunk> {
```

Comment:
Could you explain why this is not `OutputChunk`? Are you expecting `code` to not be returned sometimes?